av1an-core/src/encoder.rs

```diff
@@ -340,20 +341,40 @@ impl Encoder {
   fn replace_q(self, index: usize, q: usize) -> (usize, String) {
     match self {
       Self::aom | Self::vpx => (index, format!("--cq-level={}", q)),
-      Self::rav1e | Self::svt_av1 | Self::x265 | Self::x264 => (index + 1, format!("{}", q)),
+      Self::rav1e | Self::svt_av1 | Self::x265 | Self::x264 => (index + 1, q.to_string()),
```

Comment:
I changed this to use `format!("{}", q)` as opposed to `q.to_string()` since `format!` appears to generate less code. See this godbolt example here: https://godbolt.org/z/d4aWsr9z3

I'm open to changing it to `to_string` as well, but as far as I understand right now I think `format!` is better (not sure why they even generate different assembly in the first place)

Reply:
That is very interesting that `format` generates better assembly. I would expect format to generate worse assembly. Clippy also has a lint that should trigger when doing something like this, I'm surprised it was not. That kind of seems like something that should be reported to the Rust compiler issues.